zebra-rpc/src/methods/tests/vectors.rs

```diff
@@ -1149,31 +1149,36 @@ async fn rpc_getnetworksolps() {
 
     let get_network_sol_ps_inputs = [
         (None, None),
+        (Some(-4), None),
+        (Some(-3), Some(0)),
+        (Some(-2), Some(-4)),
+        (Some(-1), Some(10)),
+        (Some(-1), Some(i32::MAX)),
         (Some(0), None),
         (Some(0), Some(0)),
-        (Some(0), Some(-1)),
+        (Some(0), Some(-3)),
         (Some(0), Some(10)),
         (Some(0), Some(i32::MAX)),
         (Some(1), None),
         (Some(1), Some(0)),
-        (Some(1), Some(-1)),
+        (Some(1), Some(-2)),
         (Some(1), Some(10)),
         (Some(1), Some(i32::MAX)),
-        (Some(usize::MAX), None),
-        (Some(usize::MAX), Some(0)),
-        (Some(usize::MAX), Some(-1)),
-        (Some(usize::MAX), Some(10)),
-        (Some(usize::MAX), Some(i32::MAX)),
+        (Some(i32::MAX), None),
+        (Some(i32::MAX), Some(0)),
+        (Some(i32::MAX), Some(-1)),
+        (Some(i32::MAX), Some(10)),
+        (Some(i32::MAX), Some(i32::MAX)),
     ];
 
     for (num_blocks_input, height_input) in get_network_sol_ps_inputs {
         let get_network_sol_ps_result = get_block_template_rpc
             .get_network_sol_ps(num_blocks_input, height_input)
             .await;
         assert!(
-            get_network_sol_ps_result
-                .is_ok(),
-            "get_network_sol_ps({num_blocks_input:?}, {height_input:?}) call with should be ok, got: {get_network_sol_ps_result:?}"
+            get_network_sol_ps_result.is_ok(),
```

Comment:
This error shouldn't happen, maybe populated_state() needs to wait for the block writes to complete:
> Message:  get_network_sol_ps(Some(-3), Some(0)) call should be ok, got: Err(Error { code: ServerError(0), message: "No blocks in state", data: None })

https://github.com/ZcashFoundation/zebra/actions/runs/6346254471/job/17239516115?pr=7647
https://github.com/ZcashFoundation/zebra/actions/runs/6346254514/job/17239704647?pr=7647#step:4:3169

Reply:
This was a real bug in the new code.